extensions/cdc-postgres/src/test/java/com/hazelcast/jet/cdc/postgres/MultiTableCacheIntegrationTest.java

```diff
@@ -0,0 +1,314 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.cdc.postgres;
+
+import com.hazelcast.jet.JetInstance;
+import com.hazelcast.jet.Job;
+import com.hazelcast.jet.cdc.ChangeRecord;
+import com.hazelcast.jet.cdc.Operation;
+import com.hazelcast.jet.cdc.ParsingException;
+import com.hazelcast.jet.cdc.RecordPart;
+import com.hazelcast.jet.cdc.impl.ChangeRecordImpl;
+import com.hazelcast.jet.function.TriFunction;
+import com.hazelcast.jet.pipeline.Pipeline;
+import com.hazelcast.jet.pipeline.Sinks;
+import com.hazelcast.jet.pipeline.StreamSource;
+import com.hazelcast.jet.pipeline.StreamStage;
+import com.hazelcast.map.EntryProcessor;
+import org.jetbrains.annotations.NotNull;
+import org.junit.Test;
+
+import java.io.Serializable;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.Statement;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static com.hazelcast.jet.impl.util.ExceptionUtil.rethrow;
+
+public class MultiTableCacheIntegrationTest extends AbstractPostgresCdcIntegrationTest {
+
+    private static final int MAX_CONCURRENT_OPERATIONS = 1;
+    private static final String CACHE = "cache";
+    private static final int REPEATS = 1000;
+
+    @Test
+    public void ordersOfCustomers() throws Exception {
+        StreamSource<ChangeRecord> source = sourceBuilder("source")
+                .setTableWhitelist("inventory.customers", "inventory.orders")
+                .build();
+
+        Pipeline pipeline = Pipeline.create();
+        StreamStage<ChangeRecord> allRecords = pipeline.readFrom(source)
+                .withNativeTimestamps(0);
+
+        allRecords.filter(r -> r.table().equals("customers"))
+                .apply(this::fixOrdering)
+                .writeTo(Sinks.mapWithEntryProcessor(MAX_CONCURRENT_OPERATIONS, CACHE,
+                        record -> (Integer) record.key().toMap().get("id"),
+                        CustomerEntryProcessor::new
+                ));
+
+        allRecords.filter(r -> r.table().equals("orders"))
+                .apply(this::fixOrdering)
+                .writeTo(Sinks.mapWithEntryProcessor(MAX_CONCURRENT_OPERATIONS, CACHE,
+                        record -> (Integer) record.value().toMap().get("purchaser"),
+                        OrderEntryProcessor::new
+                ));
```

Comment:
Will this work when loading larger amount of data from scratch?

Reply:
I don't know.